cirq/contrib/routing/greedy_test.py

```diff
@@ -20,10 +22,47 @@
 
 
 def test_bad_args():
+    """Test zero valued arguments in greedy router."""
     circuit = cirq.testing.random_circuit(4, 2, 0.5, random_state=5)
     device_graph = ccr.get_grid_device_graph(3, 2)
     with pytest.raises(ValueError):
         route_circuit_greedily(circuit, device_graph, max_search_radius=0)
 
     with pytest.raises(ValueError):
         route_circuit_greedily(circuit, device_graph, max_num_empty_steps=0)
+
+
+def create_circuit_and_device():
+    """Construct a small circuit and a device with line connectivity
+    to test the greedy router. This instance hangs router in Cirq 8.2.
+    """
+    num_qubits = 6
+    gate_domain = {cirq.ops.CNOT: 2}
+    circuit = cirq.testing.random_circuit(num_qubits,
+                                          15,
+                                          0.5,
+                                          gate_domain,
+                                          random_state=37)
+    device_graph = ccr.get_linear_device_graph(num_qubits)
+    return circuit, device_graph
+
+
+def create_hanging_routing_instance(circuit, device_graph):
+    """Create a test problem instance."""
+    route_circuit_greedily(circuit,
+                           device_graph,
+                           max_search_radius=2,
+                           random_state=1)
+
+
+def test_router_hanging():
+    """Run a separate process and check if greedy router hits timeout (1s)."""
+    circuit, device_graph = create_circuit_and_device()
+    process = Process(target=create_hanging_routing_instance,
+                      args=[circuit, device_graph])
+    process.start()
+    process.join(timeout=1)
+    alive = process.is_alive()
+    if alive:
+        process.terminate()
```

Comment:
I would do something like this instead - this takes care of the coverage issue as well

```
    try:
        assert not process.is_alive(), "Greedy router timeout"
    finally:
        process.terminate()
```

Reply:
Cool, thanks! Was happy to contribute. It looks like all tests passed now.:)